pkg/kubecost/summaryallocation.go

```diff
@@ -1170,6 +1170,101 @@ func (sas *SummaryAllocationSet) TotalCost() float64 {
 	return tc
 }
 
+// func to calculate average RAM efficiency over SummaryAllocationSet
+func (sas *SummaryAllocationSet) RAMEfficiency() float64 {
+	if sas == nil {
+		return 0.0
+	}
+
+	sas.RLock()
+	defer sas.RUnlock()
+
+	totalRAMBytesUsage := 0.0
+	totalRAMBytesRequest := 0.0
+	for _, sa := range sas.SummaryAllocations {
+		if sa.RAMBytesRequestAverage > 0 {
+			totalRAMBytesUsage += sa.RAMBytesUsageAverage
+			totalRAMBytesRequest += sa.RAMBytesRequestAverage
+		} else {
+			if sa.RAMBytesUsageAverage == 0.0 && sa.RAMCost == 0.0 {
+				totalRAMBytesUsage += 1.0
+				totalRAMBytesRequest += 1.0
+			}
+		}
+	}
+
+	if totalRAMBytesUsage > 0 {
+		return totalRAMBytesUsage / totalRAMBytesRequest
+	}
```

Comment:
I believe this should actually be:
```go
if totalRAMBytesRequest > 0 {
```

As a divide-by-zero check.

Reply:
I checked this and edited both RAM and CPU efficiency.